presto-druid/src/main/java/com/facebook/presto/druid/DruidFilterExpressionConverter.java

```diff
@@ -185,7 +185,7 @@ public DruidExpression visitCall(CallExpression call, Function<VariableReference
                 throw new PrestoException(DRUID_PUSHDOWN_UNSUPPORTED_EXPRESSION, "Invalid Timestamp literal in Druid filter: " + argument.toString());
             }
             SqlTimestamp value = new SqlTimestamp((long) ((ConstantExpression) argument).getValue());
-            return new DruidExpression("'" + value.toString() + "'", Origin.LITERAL);
+            return derived(" TIMESTAMP '" + value.toString() + "'");
```

Comment:
Can I have the query you are using for test?   For the query below I ran, this 'if' statement is not reachable.
`select * from druid.wikipedia where __time > timestamp '2016-06-26 19:00:00.000';`
Because for the call 'greater_than(__time, timestamp)', it will go through line 177.  And the second argument of this call is a constant which will go to the 'visitConstant' method and then call the getLiteralAsString

Reply:
Hi, 
I use the following test case for original code before I change:
`SELECT * FROM druid.druid.wikipedia where __time > timestamp '2015-09-12 00:00:00.000' limit 1;`
You could check following plan:
<img width="516" alt="Screen Shot 2020-08-07 at 11 11 23 AM" src="https://user-images.githubusercontent.com/30369486/89604885-c7fe8500-d89e-11ea-886e-5a55ee0ea3ee.png">

And will not go to line 177 as following,
<img width="1636" alt="Screen Shot 2020-08-07 at 11 09 58 AM" src="https://user-images.githubusercontent.com/30369486/89604826-a56c6c00-d89e-11ea-9833-25ad57ee8e97.png">




After the code I change:
`SELECT * FROM druid.druid.wikipedia where __time > timestamp '2015-09-12 00:00:00.000' limit 1;
`
<img width="455" alt="Screen Shot 2020-08-07 at 11 07 11 AM" src="https://user-images.githubusercontent.com/30369486/89604611-3262f580-d89e-11ea-8e26-a328d85f7249.png">

Hopefully It's helpful.

Reply:
Got it with many thanks! 
Looks like the difference is on the RowExpression.  The timestamp I got is a constant expression, but you got is a call of '$literal$timestamp' .  I will look into it. Thanks!

Reply:
have you rebased onto the most recent master?  I think some of the behavior might changed, just guessing, it could explain why we are seeing different RowExpression type.  Thanks!

Reply:
Yes, you are right. I try on release 0.238.2, the logic is still same as what I say. But on recent master, the behavior is just as what you said. I think there may be some changes in presto-main or presto-spi that does not take druid connector into consideration. I am worried that there may be other potential affects other than this one we find.

Reply:
hi, @zhenxiao, I could revert my code related to time pushed down, And hope the PR from beinan will fix it due to the change out of druid connector. Is it ok from your opinion?

Reply:
thanks for the discussion @weidongduan37 @beinan 
sure. go ahead. Let's rebased on most recent master, and revert time pushdown changes in this PR.
I will follow up with optimizer changes. My guess is RowExpression is replacing Expression recently.